DS/DS-Session2-Part1/Problem4/CounselingProcess.java

```diff
@@ -0,0 +1,159 @@
+	 * @param studentListFile - student excel file
+	 * @throws IOException
+	 */
+	public void addStudentFromExcel(File studentListFile) throws IOException{
+		Workbook workbook;
+		try {
+			workbook = Workbook.getWorkbook(studentListFile);
+			Sheet sheet = workbook.getSheet(0);
+          
+			for(int i=0; i < sheet.getRows(); i++) {
+				Cell cell1 = sheet.getCell(0, i);
+				Cell cell2 = sheet.getCell(1, i);
+				Cell cell3 = sheet.getCell(2, i);
+				String prefList[] = new String[5];
+				for(int j=0; j<5; j++) {
+					prefList[j] = sheet.getCell(j+3, i).getContents().toString();
+				}
+				Student student = new Student(cell3.getContents().toString(), Integer.parseInt(cell2.getContents().toString()),
+					Integer.parseInt(cell1.getContents().toString()), prefList);
+				this.studentList.add(student);
+		}
+		}catch(BiffException e) {
+            e.printStackTrace();
+        }
+	}
+	
+	/**
+	 * Read Program offer data from excel file
+	 * @param programListFile - Program list excel file
+	 * @throws IOException
+	 */
+	public void addProgramFromExcel(File programListFile) throws IOException{
+		Workbook workbook;
+		try {
+			workbook = Workbook.getWorkbook(programListFile);
+			Sheet sheet = workbook.getSheet(0);
+          
+			for(int i=0; i < sheet.getRows(); i++) {
+				Cell cell1 = sheet.getCell(0, i);
+				Cell cell2 = sheet.getCell(1, i);
+				Cell cell3 = sheet.getCell(2, i);
+				Program program = new Program(cell1.getContents().toString(), cell2.getContents().toString(), 
+        		  Integer.parseInt(cell3.getContents().toString()));
+				this.programList[i] = program;
+          }
+		}catch(BiffException e) {
+            e.printStackTrace();
+        }
+	}
+	
+	/**
+	 * Allocate program to student
+	 */
+	public void allocatePrograms() {
+		while(!this.studentList.isEmpty()) {
+			Student student = this.studentList.poll();
+			String prefList[] = student.getPrefProgramList();
+			boolean outerFlag = false;
+			for(int i=0; i < prefList.length; i++) {
+				for(int j=0; j<this.programList.length; j++) {
+					if( prefList[i].equals( this.programList[j].getCode() )) {
+						if(this.programList[j].getCapacity() != 0) {
+							outerFlag = true;
+							student.setProgram(this.programList[j]);
+							this.programList[j].addStudentToProgram(student);
+							this.programList[j].updateCapacity(this.programList[j].getCapacity() - 1);
+							break;
+						}	
+						else {
+							break;
+						}
+					}
+				}
+				if(outerFlag) {
+					break;
+				}
+			}
+		}
+	}
+	
+	/**
+	 * Write all allocated information to counselingList
+	 * @param counselingList - output counseling excel file
+	 */
+	public void writeDataInFile(File counselingList) {
+		WritableWorkbook myFirstWbook = null;
+        try {
+
+            myFirstWbook = Workbook.createWorkbook(counselingList);
+            WritableSheet excelSheet = myFirstWbook.createSheet("Sheet 1", 0);
+
+            int col=0, row=0;
+			for(int i=0; i<this.programList.length; i++) {
```

Comment:
The program should output the list of students along with their allocated programs